### DISCUSSION TITLE
[Question]: VPN with rotate IP address

### DISCUSSION THREAD

**Discussion by ericc-dream** [e107inc/e107#5299] [Q&A Core]
### Question

Hi 

Well everything is in the title ! 
I have one user who is using a VPN with IP address changing in a round-robin. 
So he connect to the site, he can post one or two messages, and suddenly he is disconnected because his IP address changes ! 

Now I'm experiencing the same problem from my corporate network. Seems that IT service is using several "exit point" in our network, and, randomly, I'm disconnected from my site for the same reason: IP address have changed 

This seems to be very specific to e107. I mean, I'm connected all day long to my Gmail inbox and never experience this problem. Same on other web site like github by example 

Is there a solution ? 
An alternate login method which can avoid this issue ? 

When you write a long message and posting to discover that you was disconnected and so the message is rejected and lost, believe me this is irritating 

Best regards

### Expected Source

I looked in the Admin panel but didn't found anything related to this ... but I may have missed it

**Comment by Deltik** [comment]
I suggest setting `define('e_SECURITY_LEVEL', 3);` in `./e107_config.php`.  This turns off [e107's sensitivity to changes in `$_SERVER['REMOTE_ADDR']`](https://github.com/e107inc/e107/blob/d1f1b9e7cc213882a6027a0f50010f8a21f847f5/e107_handlers/session_handler.php#L205), so IP addresses changes should no longer be considered for de-authenticating a user.

Related Gitter discussion: https://matrix.to/#/!srhnCKlMgxFeuSPXPb:gitter.im/$xXzbMoYnOxHlXjKHlh8NHKH_9p0WZQD3mQaNDNeuqfo?via=gitter.im&via=matrix.org

**Comment by ericc-dream** [comment]
Hi 
It works, but it's a little bit weird ... 
I mean : I connect to my site, I open the site "whatismyip.com" to check my IP address in another tab
When it change, I can still post some message in the chatbox which is on the main page, and the message is accepted. Which is already a big improvement 
However, if I refresh the page, with F5 (or with an add-on of Firefox to reload automatically a page after a predefined time) , I'm disconnected ... 
I think that it's more or less acceptable, as, in this case, you see it. You see that you are not connected anymore, and the chatbox is not visible anymore 

I forgot to test what's happen if I change of page ... will do that next week

**Comment by Vodhin** [comment]
You might try changing `Site Preferences  >> Security & Protection > User Tracking method` to `Cookies` instead of `Sessions`?

**Comment by ericc-dream** [comment]
Hi 
Thanks for the suggestion, but it seems that I'm already in "Cookies" with a lifetime of 86400 seconds (24hours)

**Reply by Vodhin** [reply]
I assume you tried the ` Session` option instead of `Cookies` - Try it if you have not already done that.

**Comment by ericc-dream** [comment]
yes, as you suggested to try "cookies" while I was already selected, then I changed to "session" 
I will see, actually I'm at home so can't and the user with his VPN isn't connected during the week-end neither 
We will see next week 
Thanks
